src/clients/googleAnalytics.ts

```diff
@@ -24,15 +26,23 @@ const googleAnalytics = () => {
     gtag('event', eventName, parameters);
   });
 
-  function buttonPressed<E extends ButtonEventName>(
+  function buttonPressed<E extends EventName>(
     buttonName: E,
-    parameters?: ButtonEventMap[E] extends undefined ? never : ButtonEventMap[E],
+    parameters?: EventMap[E] extends undefined ? never : EventMap[E],
   ): void {
     sendEvent('button_pressed', { button_name: buttonName, ...parameters });
   }
 
+  function error<E extends EventName>(
+    errorName: E,
+    parameters?: EventMap[E] extends undefined ? never : EventMap[E],
+  ): void {
+    sendEvent('error', { error_name: errorName, ...parameters });
+  }
```

Comment:
Won't we use something like Sentry to track errors?

Reply:
Yes, but I wasn't thinking about that in this way. Perhaps there is a better name for the information being captured here. In the `activate` function error callback they are not application errors that need to be solved. They are connection errors that could be the result of a user action (rejecting the metamask) or a UI flow (we tried to take an action before authorization. 

The events mirror toasts being shown to the user so for the this reason I thought it would be nice to capture analytis on them.

We should ticket adding sentry it currently isn't configured

Reply:
Ah I see! That makes sense.
If we could detect that an error comes from the user rejecting the connection via MetaMask, then we could send the analytic event only in that case (and its name could be something like `metamask_user_rejection`). If the error comes from the UI flow, then I say it should be reported in Sentry as it's something we have control over and should fix ourselves.

Sounds good about the ticket, I'll pick it up then.

Reply:
RE: Metamask rejection that is getting sent as you described

RE: Other errors, I think we should have a discussion about this. Sentry is great for programming errors but reporting non programming errors (errors shown to the user as toasts or input validation errors for example) gets complicated because the solution will likely involve other disciplines. Product and Marketing don't need to look at Sentry and will have heavier influence in raising the issue and proposing a solution so I think it makes sense to surface this data to them.

Reply:
Ok I see your point, it's true Sentry should only be used for errors WE can fix basically (other than changing the UX if too many users are getting frustrated).
Let's go with your current solution then, it makes sense!